src/upgrade/upgrade_scripts/5.12.0/upgrade_objectmds_old_versions.js

```diff
@@ -0,0 +1,99 @@
+/* Copyright (C) 2021 NooBaa */
+"use strict";
+
+const util = require('util');
+const Cursor = require('pg-cursor');
+
+async function run({ dbg, db_client }) {
+    let cursor;
+    // using 2 different clients due to https://github.com/brianc/node-postgres/issues/1895
+    let pg_client_cursor;
+    let pg_client;
+    try {
+        if (process.env.DB_TYPE !== 'postgres') {
+            dbg.log0('upgrade postgres objectmds latest index: not running on postgres - no upgrade needed...');
+            return;
+        }
+        dbg.log0('starting upgrade version past on non latest object versions...');
+
+        // The following upgrade process will do the following:
+        // 1. update old versions of an object with version_past = true
+        // 2. drop the current index - idx_btree_objectmds_latest_version_index
+
+        // idx_btree_objectmds_latest_version_index postgres index is incorrect since 
+        // postgres considers nulls as distinct, causes multiple latest versions of objectmds to be allowed
+        // recreation of the index is being handled lazily in md_store constructor
+
+        pg_client_cursor = await db_client.instance().pool.connect();
+        pg_client = await db_client.instance().pool.connect();
+        cursor = await db_client.instance().sql_query(pg_client_cursor,
```

Comment:
I would prefer that we move the pool connect into sql_query if possible.
Do we have to call client.release() explicitly after using it? 
For the cursor case is it enough to call close on the cursor to release back the connection to the pool?

Reply:
According to pg documentation: https://node-postgres.com/features/pooling
```
You must always return the client to the pool if you successfully check it out,
regardless of whether or not there was an error with the queries you ran on the client.
If you don't check in the client your application will leak them and 
eventually your pool will be empty forever and all future requests to check out a client from the pool will wait forever.

```
And regarding the cursor - in pg doc example they call client.release() right after cursor.close() so closing cursor will not release the connection back to the pool.

Also, the update query is inside a loop so moving the pool.connect() and client.release() to sql_query() will cause extra work of acquiring and releasing a connection for each iteration. On the other hand, maybe keeping the same connection for all the iterations
might be a worse behavior. WDYT? @guymguym

Reply:
ok in that case I would prefer to have a separate method that would encapsulate it inside the pg client e.g:

```js
async sql_query_batch(query, values, batch_size, batch_handler) {
  let client;
  try {
    client = await this.pool.connect();
    let cursor;
    try {
      cursor = client.query(new Cursor(query, values));
      for (;;) {
        const rows = await cursor.read(batch_size);
        if (!rows.length) break;
        const should_continue = await batch_handler(rows);
        if (!should_continue) break;
      }
    } finally {
      if (cursor) await cursor.close();
    }
  } finally {
    if (client) await client.release();
  }
}

```

Reply:
sounds good, I updated the PR and left the update queries pg_client in the upgrade script instead of creating it and releasing it on every update.